### PR TITLE
Klasa: SG v2.2.0

### PR DESCRIPTION
**Pull request by AdityaTD**
### Description of Your Pull Request
This PR is currently using Klasa's SGv2.2.0 branch and as soon as that branch is merged with master, this PR will be accepted.

### Changes Proposed and or Files Changed
- Updates RethinkDB provider to work with SGv2.2.0
- Added New Schema files and removed old ones
- Fixed major bugs with commands and monitors
- Added Caching for Global Ranks, etc.

### Label Your Pull Request
- [x] This PR modifies existing code to enhance the bot
- [x] This PR fixes a bug within the bot
- [x] This PR removes/renames methods or properties
- [x] This PR adds methods, features, commands or properties

### CODE REVIEW

File: structures/PenguClient.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const { Client } = require("klasa");
 const { Client: IdioticAPI } = require("idiotic-api");
-const permissionLevels = require("./PermissionLevels");
+const permissionLevels = require(`./permissionLevels`);
```
Comment (MrJacz) [inline structures/PenguClient.js]: thonk

File: providers/rethinkdb.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
-const { Provider } = require("klasa");
+const { Provider, util: { mergeObjects } } = require("klasa");
 const rethink = require("rethinkdbdash");
 
 module.exports = class extends Provider {
 
     constructor(...args) {
         super(...args);
-        this.db = rethink(this.client.options.providers.rethinkdb || { db: "pengubot" });
+        this.db = rethink(this.client.options.providers.rethinkdb || { db: "pengubot", timeout: 100 });
```
Comment (MrJacz) [inline providers/rethinkdb.js]: What does timeout do?

File: package.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
     "fast-html-parser": "^1.0.1",
     "google-translate-api": "^2.3.0",
     "idiotic-api": "^0.2.8",
-    "klasa": "github:dirigeants/klasa",
+    "klasa": "github:dirigeants/klasa#SGv2.2.0",
```
Comment (MrJacz) [inline package.json]: if SG v2.2 gets merged this wont work.

File: providers/rethinkdb.js (hunk 2 above)
Comment (MrJacz) [reply providers/rethinkdb.js]: Lmfao, `timeout` is in seconds. no wonder why you get errors with sharding

File: package.json (hunk 3 above)
Comment (MrJacz) [reply package.json]: Yes but... if you're going to merge this into rewrite when SGv2.2 is merged then it creates more work for you when you could just remove the branch and have it ready to be merged.

File: providers/rethinkdb.js (hunk 2 above)
Comment (AdityaTD) [reply providers/rethinkdb.js]: Not really, when I had ot set to default aka 20 seconds it showed the error I sent yesterday.

File: package.json (hunk 3 above)
Comment (AdityaTD) [reply package.json]: Changing a dependency isn't more work. I have it this way so I can test.

Comment (MrJacz) [reply package.json]: Making a commit = work. but okay

File: providers/rethinkdb.js (hunk 2 above)
Comment (MrJacz) [reply providers/rethinkdb.js]: then go blame @kyranet

Comment (kyranet) [reply providers/rethinkdb.js]: What do I have to do with RethinkDB, @MrJacz? 😕

Comment (kyranet) [inline providers/rethinkdb.js]: You should revert this change and set up the database configuration in your `KlasaClientOptions`:

```javascript
new KlasaClient({
    // ...
    providers: {
        default: 'rethinkdb',
        rethinkdb: { db: 'pengubot', timeout: 100 }
    }
    // ...
});
```

Comment (MrJacz) [reply providers/rethinkdb.js]: @kyranet The timeout issue yukine was having with SG

Comment (AdityaTD) [reply providers/rethinkdb.js]: Okay will do, thanks.
